File: 5 - Form Template Method/JS/CapitalStrategyRevolver.js
import CapitalStrategy from "./CapitalStrategy";

export default class CapitalStrategyRevolver extends CapitalStrategy{
    capital(loan) {
        return (loan.getOutstandingRiskAmount() * this.duration(loan) * this.riskFactor(loan)) +
            (loan.unusedRiskAmount() * this.duration(loan) * this.unusedRiskFactor(loan));
    }
}

File: 5 - Form Template Method/C#/MPG.FormTemplateMethod.After/CapitalStrategyRevolver.cs
﻿namespace MPG.FormTemplateMethod.After
{
    public class CapitalStrategyRevolver : CapitalStrategy
    {
        public override double Capital(Loan loan)
        {
            return Outstanding(loan) + Unused(loan);
        }

        private double Outstanding(Loan loan)
        {
            return loan.OutstandingRiskAmount() * Duration(loan) * RiskFactor(loan);
        }

        private double Unused(Loan loan)
        {
            return loan.UnusedRiskAmount() * Duration(loan) * UnusedRiskFactor(loan);
        }
    }
}